src/autoscaler/api/broker/broker.go

```diff
@@ -507,7 +508,13 @@ func (b *Broker) Bind(ctx context.Context, instanceID string, bindingID string,
 
 	policy, err := b.getPolicyFromJsonRawMessage(policyJson, instanceID, details.PlanID)
 	if err != nil {
-		logger.Error("get-default-policy", err)
+		logger.Error("get-default-policy: no user defined policy was provided", err)
+		return result, err
+	}
+	customMetricsCredentialType := b.conf.CustomMetricsCredentialType
+	credentialType, err := getOrDefaultCredentialType(policyJson, customMetricsCredentialType, logger)
+	if err != nil {
+		logger.Error("getOrDefaultCredentialType %w", err)
```

Comment:
As you provided the logger to `getOrDefaultCredentialType` we should be able to delegate the responsibility for logging to it, and this would just duplicate the error log.

```suggestion
```